spyder/widgets/findreplace.py

```diff
@@ -571,12 +571,18 @@ def replace_find_selection(self, focus_replace_text=False):
             replacement = re_pattern.sub(replace_text, selected_text)
             if replacement != selected_text:
                 cursor = self.editor.textCursor()
+                start_pos = cursor.selectionStart()
                 cursor.beginEditBlock()
                 cursor.removeSelectedText()
                 if not self.re_button.isChecked():
                     replacement = re.sub(r'\\(?![nrtf])(.)', r'\1', replacement)
                 cursor.insertText(replacement)
+                # Restore selection
+                self.editor.set_cursor_position(start_pos)
+                for c in range(len(replacement) - replacement.count(self.editor.get_line_separator())):

```

Comment:
This line is too long. Please define a new variable and assign it the value of `len(replacement) - replacement.count(self.editor.get_line_separator())` to avoid that.